SignalR_Server/Models/M_GameState.cs

```diff
@@ -70,72 +73,19 @@ private string TemporaryIdGenerator()
         #endregion
 
         #region Helper Methods
-
-        //THIS IS TERRIBLE, CHANGE LATER
-        public M_AnswerStats GetAnswerStats()
-        {
-            int i = 0;
-            while (GameAnswerStats[i].GameRound != GameRound || !GameAnswerStats[i].FocusedPlayerAnswer.PlayerId.Equals(FocusedPlayerId))
-            {
-                i++;
-            }
-
-            return GameAnswerStats[i];
-        }
-
-        //THIS IS TERRIBLE, CHANGE LATER
-        public void UpdateAnswerStats(M_AnswerStats newAnswerStats)
-        {
-            int i = 0;
-            while (GameAnswerStats[i].GameRound != GameRound || !GameAnswerStats[i].FocusedPlayerAnswer.PlayerId.Equals(FocusedPlayerId))
-            {
-                i++;
-            }
-
-            GameAnswerStats[i] = newAnswerStats;
-        }
-        
-        //THIS IS TERRIBLE, CHANGE LATER
-        public M_QuestionCard GetFocusedQuestion()
-        {
-            int i = 0;
-            while (!GameQuestions[i].QuestionId.Equals(FocusedQuestionId))
-            {
-                i++;
-            }
-
-            return GameQuestions[i];
-        }
-
+                
         public void GenerateNextQuestion()
         {
-            int i = new Random().Next(0, GameQuestions.Count);
-
-            while (GameQuestions[i].GameRound != GameRound)
-            {
-                i = new Random().Next(0, GameQuestions.Count);
-            }
+            FocusedQuestionId = GameQuestions[QuestionCount].QuestionId;
+            QuestionCount++;
```

Comment:
I would refactor this var name to FocusedQuestionIndex. Not a huge issue but I thought this was the max count for a second, not the current index.

Reply:
Yeah that makes sense, I'll change that on my next branch.